frontend/online-shop/src/App.tsx

```diff
@@ -0,0 +1,31 @@
+import { CssBaseline } from "@material-ui/core";
+import { ThemeProvider } from "@material-ui/styles";
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Redirect,
+} from "react-router-dom";
+import "./App.css";
+import { ProductDetail } from "./components/ProductDetail";
+import ProductList from "./components/ProductList";
+import MyTheme from "./theme/MyTheme";
+
+export const App = () => {
+  return (
+    <ThemeProvider theme={MyTheme}>
+      <CssBaseline>
+        <Router>
+          <Switch>
```

Comment:
What if the user is directly accessing a path that you don't have in your switch? For example http://localhost:4200/asdfg. For such cases you need to have a wildcard route that catches all the paths that do not fit with any other declared route and return a 404 page.

Reply:
Fixed.